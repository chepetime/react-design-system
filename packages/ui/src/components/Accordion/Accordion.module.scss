@use "./../../styles" as styles;

:root {
  --accordion-background-color: #{styles.$pink-500};
}

:global .dark {
  --accordion-background-color: #{styles.$indigo-500};
}

.Accordion {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--accordion-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
