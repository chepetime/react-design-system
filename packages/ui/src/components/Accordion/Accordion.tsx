import React from "react";
import classnames from "classnames";
import css from "./Accordion.module.scss";

export interface AccordionProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Accordion = React.forwardRef<HTMLDivElement, AccordionProps>(
  (props: AccordionProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Accordion)} ref={ref}>
        <p>Accordion component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Accordion.displayName = "Accordion";
