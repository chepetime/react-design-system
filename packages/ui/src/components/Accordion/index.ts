import {
  Accordion,
  AccordionItem,
  AccordionTrigger,
  AccordionContent,
} from "./Accordion";
import { AccordionProps } from "./Accordion";

const Components = {
  Accordion,
  AccordionItem,
  AccordionTrigger,
  AccordionContent,
};

export default Components;
export type { AccordionProps };
