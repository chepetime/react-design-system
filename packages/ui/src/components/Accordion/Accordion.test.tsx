
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { Accordion } from "./Accordion";
import { render, screen } from "@testing-library/react";

describe("Accordion", () => {
  it("renders without errors", () => {
    render(<Accordion />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<Accordion>Some children</Accordion>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<Accordion>Some children</Accordion>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<Accordion />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
