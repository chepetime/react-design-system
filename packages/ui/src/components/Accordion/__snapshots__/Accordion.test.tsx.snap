// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion matches snapshot 1`] = `
<div>
  <div
    class="Accordion"
  >
    <p>
      Accordion component working!
    </p>
    <div />
  </div>
</div>
`;
